extra/modules/live-intent-omni-channel-identity/src/test/java/org/prebid/server/hooks/modules/liveintent/omni/channel/identity/v1/LiveIntentOmniChannelIdentityProcessedAuctionRequestHookTest.java

```diff
@@ -60,15 +76,178 @@ public void setUp() {
         given(properties.getIdentityResolutionEndpoint()).willReturn("invalid_url");
         assertThatIllegalArgumentException().isThrownBy(() ->
                 new LiveIntentOmniChannelIdentityProcessedAuctionRequestHook(
-                        properties, MAPPER, httpClient, 0.01d));
+                        properties, userFpdActivityMask, MAPPER, httpClient, 0.01d));
+    }
+
+    @Test
+    public void geoPassingRestrictionShouldBeRespected() {
+        // given
+        final Geo givenGeo = Geo.builder()
+                .lat(52.51671856406936f)
+                .lon(13.377639726342583f)
+                .city("Berlin")
+                .country("Germany")
+                .build();
+        final Device givenDevice = Device.builder()
+                .geo(givenGeo)
+                .ip("192.168.127.12")
+                .ifa("foo")
+                .macsha1("bar")
+                .macmd5("baz")
+                .dpidsha1("boo")
+                .dpidmd5("far")
+                .didsha1("zoo")
+                .didmd5("goo")
+                .build();
+        final BidRequest givenBidRequest = BidRequest.builder().id("request").device(givenDevice).build();
+
+        final Geo expectedGeo = givenGeo.toBuilder()
+                .country(null)
+                .city(null)
+                .lat(52.52f)
+                .lon(13.38f)
+                .build();
+        final Device expectedDevice = givenDevice.toBuilder()
+                .geo(expectedGeo)
+                .ip("192.168.127.0")
+                .ifa(null)
+                .macsha1(null)
+                .macmd5(null)
+                .dpidsha1(null)
+                .dpidmd5(null)
+                .didsha1(null)
+                .didmd5(null)
+                .build();
+        final BidRequest expectedBidRequest = givenBidRequest.toBuilder().device(expectedDevice).build();
+
+        final Eid expectedEid = Eid.builder().source("liveintent.com").build();
+
+        final String responseBody = MAPPER.encodeToString(IdResResponse.of(List.of(expectedEid)));
+        given(httpClient.post(any(), any(), any(), anyLong()))
+                .willReturn(Future.succeededFuture(HttpClientResponse.of(200, null, responseBody)));
+
+        final ActivityController activityController = ActivityController.of(
+                false, List.of(), activityInfrastructureDebug);
+
+        final ActivityInfrastructure givenActivityInfrastructure = new ActivityInfrastructure(
+                Map.of(Activity.TRANSMIT_TID, activityController,
+                        Activity.TRANSMIT_UFPD, activityController),
+                activityInfrastructureDebug);
+
+        final AuctionInvocationContext auctionInvocationContext = AuctionInvocationContextImpl.of(
+                null,
+                AuctionContext.builder().activityInfrastructure(givenActivityInfrastructure).build(),
+                false,
+                null,
+                null);
```

Comment:
Mock entire activity infrastructure. Same for other tests.
```
@Mock
private ActivityInfrastructure activityInfrastructure;
```

Reply:
Done in https://github.com/prebid/prebid-server-java/pull/4185/commits/3a0d8ececb71e95faee4c47556eb4e5d4715b94e